bqplot/traits.py

```diff
@@ -135,13 +135,33 @@ def _to_json(a, obj=None):
             return {'values': a, 'type': None}
 
     def _set_value(self, value):
-        if value is None or len(value) == 0:
-            return np.asarray(value)
-        if(isinstance(value, np.ndarray) and np.issubdtype(value.dtype, np.datetime64)):
+        if isinstance(value, np.ndarray):
             return value
+        elif value is None or len(value) == 0:
+            return np.asarray(value)
         else:
             return np.asarray(value)
 
+    def set(self, obj, value):
+        #TODO: We shouldnt have to overload the set because
+        # numpy doesn't support == comparions
+        new_value = self._validate(obj, value)
+        try:
+            old_value = obj._trait_values[self.name]
+        except KeyError:
+            old_value = self.default_value
+
+        obj._trait_values[self.name] = new_value
+        try:
+            silent = np.array_equal(old_value, new_value)
+        except:
+            # if there is an error in comparing, default to notify
+            silent = False
+        if silent is not True:
```

Comment:
we can probably simplify this into `if not np.array_equal(old_value, new_value)`